routes/index.js

```diff
@@ -1,14 +1,50 @@
 var express = require('express');
 var router = express.Router();
-var {User, Post} = require('../db/models');
+var {User, Post,Follow} = require('../db/models');
+const { loginUser, restoreUser } = require("../auth");
 /* GET home page. */
-router.get('/', async function(req, res, next) {
+router.get('/', restoreUser,async function(req, res, next) {
+  /*
+  To get users with an array of followers
+  const getUsers = await User.findAll({
+    include: {
+      model: User,
+      as:"Followers"
+    }
+  });
+  to get users with an array of following
+  const getUsers = await User.findAll({
+    include: {
+      model: User,
+      as:"Followers"
+    }
+  });
+  */
   const sixUsers = await Post.findAll({
     include: User,
     limit:6
   });
+  if(!res.locals.authenticated){
+    res.render('index', { 
+      title: 'CodeX is a place to write, read, and connect',
+      post:sixUsers
+    });
+  }else{
+    const currUser = await User.findOne({
+      where: {
+        id: req.session.auth.userId
+      }
+    })
```

Comment:
The user is already added to `res.locals` with `res.locals.user` when using the `restoreUser` middleware. You do not need to query for the user again.